next/src/field/schema.ts

```diff
@@ -32,12 +32,60 @@ function getJsonType(schema: NonBooleanJsfSchema): string {
   return 'text'
 }
 
+/**
+ * Convert options to the required format
+ */
+function convertToOptions(nodeOptions: any[]): { label: string, value: any }[] {
+  return nodeOptions
+    .filter(option => option !== null)
+    .map(({ title, const: value, 'x-jsf-presentation': presentation, ...item }) => {
+      // Extract meta from x-jsf-presentation if it exists
+      const meta = presentation?.meta
+
+      return {
+        label: title,
+        value,
+        // Include meta at the root level if it exists
+        ...(meta && { meta }),
+        ...item,
+      }
+    })
+}
+
+/**
+ * TODO: go over the added code here and make sure it's correct and includes all the necessary fields
+ * this is a rough draft and needs to be cleaned up
+ */
+
+/**
+ * Get field options from schema
+ */
+function getFieldOptions(schema: NonBooleanJsfSchema, presentation: any) {
+  // Handle deprecated presentation.options first
+  if (presentation.options) {
+    return presentation.options
+  }
+
+  // Handle oneOf or radio input type
+  if (schema.oneOf || presentation.inputType === 'radio') {
```

Comment:
Based on the comment above https://github.com/remoteoss/json-schema-form/pull/141/files#r1968082432, it means this does not need to look into the `inputType`